src/libponyc/pass/syntax.c

```diff
@@ -696,14 +696,19 @@ static ast_result_t syntax_consume(pass_opt_t* opt, ast_t* ast)
   {
     case TK_THIS:
     case TK_REFERENCE:
-    case TK_DOT:
       return AST_OK;
-
+    case TK_DOT: {
+        AST_GET_CHILDREN(term, left, right);
+        if (ast_id(left) != TK_CALL && ast_id(left) != TK_SEQ) 
```

Comment:
Here  I went a bit too defensive and raising errors for known errors. 
Mainly because, by allowing other illegal constructs, we get more detailed/friendlier errors down the line where we do lot more tree traversal anyway.
 
Do you suggest we go a bit more aggressive here? It could mean we will have to do go up and down the ast a bit more.